metrics/ton.py

```
@@ -99,6 +100,63 @@ def get_params_from_state(state_data: dict) -> dict:
+
+    @staticmethod
+    def validate_state(state_data: dict[str, Any]) -> bool:
+        """Require the bucketed-seqno anchor in state data."""
+        return bool(state_data and state_data.get("agreement_anchor"))
+
+    @staticmethod
+    def get_params_from_state(state_data: dict[str, Any]) -> dict[str, Any]:
+        """Return parameters for the account-state lookup."""
+        return {"address": _AGREEMENT_PROBE_ADDRESS}
+
+    def __init__(self, *args: Any, **kwargs: Any) -> None:
```

Ruff (0.15.13)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `*args`

(ANN401)